src/components/Property/RaiseWorkOrder/RateScheduleItemView.js

```diff
@@ -18,12 +18,24 @@ const RateScheduleItemView = ({
   setSorCodeArrays,
   sorSearchRequest,
   setPageToMultipleSORs,
+  formState,
 }) => {
   const [
     arrayOfRateScheduleItemComponentIndexes,
     setArrayOfRateScheduleItemComponentIndexes,
   ] = useState(sorCodeArrays.map((v, index) => index))
 
+  useEffect(() => {
+    if (arrayOfRateScheduleItemComponentIndexes.length > 1) {
+      if (
+        formState.rateScheduleItems.length <
+        arrayOfRateScheduleItemComponentIndexes.length
+      ) {
+        arrayOfRateScheduleItemComponentIndexes.pop()
+      }
+    }
+  }, arrayOfRateScheduleItemComponentIndexes)
+
```

Comment:
This could be much cleaner if you negated the if statements. For example:

```suggestion
  useEffect(() => {
    if (arrayOfRateScheduleItemComponentIndexes.length === 0) return;
    if (formState.rateScheduleItems.length === arrayOfRateScheduleItemComponentIndexes.length) return;
      
    arrayOfRateScheduleItemComponentIndexes.pop();
  }, arrayOfRateScheduleItemComponentIndexes)

```

Reply:
To my eyes, I think the way Dave wrote it is easier to read but that's subjective

Reply:
Also, the logic is different. In the top example if arrayOfRateScheduleItemComponentIndexes.length > formState.rateScheduleItems.length, the pop won't happen, whereas in the reworked example it will